### ISSUE TITLE
Ability to configure Target implementation used by FeignClientFactoryBean

### ISSUE THREAD

**Issue by xyloman** [spring-cloud/spring-cloud-netflix#287]
We have a few use cases where a custom Target implementation would be necessary to gain access to the Request to add a header.  We found that using the feign.RequestInterceptor had a short coming in that it does not expose the target url that is either load balanced to or hard coded until after the apply is executed line 164 in the SynchronousMethodHandler.  We are requiring the full URL in order to generate an OAuth 1.0a secured request to populate the Authorization header.   We were able to work around this issue by using the Feign builder with a custom implementation of Target (see https://github.com/Netflix/feign#multiple-interfaces) explicitly instead of the annotation + FeignClientFactoryBean.  I am not completely familiar with the FeignClientFactoryBean but it does appear that it could expose an additional autowired bean for the target to use but I might be over looking a different approach.  I found a conversation on the feign github that is similar https://github.com/Netflix/feign/issues/214.

**Comment by spencergibb** [comment]
I could see this combined with #288 (ie customize the Feign environment per client.)  Though, by overriding the target you would potentially loose ribbon integration.

**Comment by xyloman** [comment]
I agree, but if the consumer is taking the step to customize I assume that they would do their homework to maintain the ribbon integration in their custom Target implementation.
